src/coreclr/vm/dllimportcallback.cpp

```diff
@@ -304,20 +304,20 @@ VOID __fastcall UMEntryThunk::ReportViolation(UMEntryThunk* pEntryThunk)
     MethodDesc* pMethodDesc = pEntryThunk->GetMethod();
 
     SString namespaceOrClassName;
-    SString methodName;
-    SString moduleName;
+    EString<EncodingUTF8> methodName;
+    EString<EncodingUTF8> moduleName;
 
     pMethodDesc->GetMethodInfoNoSig(namespaceOrClassName, methodName);
-    moduleName.SetUTF8(pMethodDesc->GetModule()->GetSimpleName());
+    moduleName.Set(pMethodDesc->GetModule()->GetSimpleName());
 
     SString message;
 
-    message.Printf(W("A callback was made on a garbage collected delegate of type '%s!%s::%s'."),
-        moduleName.GetUnicode(),
-        namespaceOrClassName.GetUnicode(),
-        methodName.GetUnicode());
+    message.Printf(W("A callback was made on a garbage collected delegate of type '%S!%s::%S'."),
```

Comment:
I believe that Printf used current Ansi encoding for `%S` that won't match UTF8 string passed in.